python/semantic_kernel/connectors/memory/redis/redis_memory_store.py

```diff
@@ -0,0 +1,530 @@
+            * If the record already exists, it will be updated.
+            * If the record does not exist, it will be created.
+
+        Note: if the records do not have the same dimensionality configured for the collection,
+        they will not be detected to belong to the collection in Redis.
+
+        Arguemnts:
+            collection_name {str} -- Name for a collection of embeddings
+            records {List[MemoryRecord]} -- List of memory records to upsert
+
+        Returns
+            List[str] -- Redis keys associated with the upserted memory records, or an empty list if an error occured
+        """
+
+        if not await self.does_collection_exist_async(collection_name):
+            self._logger.error(f'Collection "{collection_name}" does not exist')
+            raise Exception(f'Collection "{collection_name}" does not exist')
+
+        keys = list()
+        for rec in records:
+            # Typical Redis key structure: collection_name:{some identifier}
+            rec._key = redis_key(collection_name, rec._id)
+
+            # Overwrites previous data or inserts new key if not present
+            # Index registers any hash matching its schema and prefixed with collection_name:
+            self._database.hset(
+                rec._key,
+                mapping=self._serialize_record_to_redis(rec),
+            )
+            keys.append(rec._key)
+
+        return keys
+
+    async def get_async(
+        self, collection_name: str, key: str, with_embedding: bool = False
+    ) -> MemoryRecord:
+        """
+        Gets a memory record from the data store. Does not guarantee that the collection exists.
+
+        Arguments:
+            collection_name {str} -- Name for a collection of embeddings
+            key {str} -- ID associated with the memory to get
+            with_embedding {bool} -- Include embedding with the memory record, default to False
+
+        Returns:
+            MemoryRecord -- The memory record if found, else None
+        """
+
+        batch = await self.get_batch_async(collection_name, [key], with_embedding)
+        return batch[0] if len(batch) else None
+
+    async def get_batch_async(
+        self, collection_name: str, keys: List[str], with_embeddings: bool = False
+    ) -> List[MemoryRecord]:
+        """
+        Gets a batch of memory records from the data store. Does not guarantee that the collection exists
+
+        Arguments:
+            collection_name {str} -- Name for a collection of embeddings
+            keys {List[str]} -- IDs associated with the memory records to get
+            with_embedding {bool} -- Include embeddings with the memory records, default to False
+
+        Returns:
+            List[MemoryRecord] -- The memory records if found, else an empty list
+        """
+        if not await self.does_collection_exist_async(collection_name):
+            raise Exception(f'Collection "{collection_name}" does not exist')
```

Comment:
In some places where collection does not exist, there is also logging. Looks like it's missing here.
```suggestion
            self._logger.error(f'Collection "{collection_name}" does not exist')
            raise Exception(f'Collection "{collection_name}" does not exist')
```